erfa/ufunc.c.templ

```diff
@@ -10,12 +10,17 @@
  */
 
 #define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION
+#define Py_LIMITED_API 0x030900f0
 #include "Python.h"
 #include "numpy/arrayobject.h"
 #include "numpy/ufuncobject.h"
 #include "erfa.h"
 #include "erfaextra.h"
 
+struct _typeobject {
+    int dummy;
```

Comment:
Windows complains if there are no members in the struct - the _typeobject struct is a workaround for old versions of GCC.

Reply:
Can we insist on a newer gcc? We're at gcc-13 now... I'm not quite sure what the constraints are (nor why this was not necessary before).

Reply:
Let me try and see which ones fail if I remove

Reply:
Ok so the mussllinux image doens't have a recent compiler, as well as the ubuntu-20.04 one used for oldestdeps. This is a pretty simple workaround so I think it's worth keeping rather than try and mess around with updating compilers in CI?

Reply:
Still confused that it didn't fail before, but I agree that the work-around is simple, and you have included a comment that will allow us to remove it eventually, so that's fine.

Reply:
This is only an issue when enabling the limited API, see https://github.com/numpy/numpy/issues/16970 for more details